src/messaging/ExchangeContext.h

```diff
@@ -199,6 +182,24 @@ class DLL_EXPORT ExchangeContext : public ReliableMessageContext,
     void Free();
     void Reset();
 
+    /**
+     *  Determine whether a response is expected for messages sent over
+     *  this exchange.
+     *
+     *  @return Returns 'true' if response expected, else 'false'.
+     */
+    bool IsResponseExpected() const;
+
+    /**
+     *  Set whether a response is expected on this exchange based on if flag kExpectResponse
+     *  is set or not in sendFlags when send a message over exchange.
```

Comment:
I don't quite understand the setup here.  If I send a message with the `kExpectResponse` set, then this will get called?  And we will be expecting a response until .... when?  What if other messages with other values of `kExpectResponse` are sent in the meantime?

Reply:
Yes, if you send a message with the kExpectResponse set,   SetResponseExpected(true) will be called, and you will be expecting a response until timeout set by SetResponseTimeout(Timeout timeout).

Each exchange allows only one 'response expected' message be outstanding at a time, if exchange receive another message with kExpectResponse, SendMessage will return error CHIP_ERROR_INCORRECT_STATE